internal/ui/ui_glfw.go

```diff
@@ -860,10 +987,38 @@ func (u *userInterfaceImpl) initOnMainThread(options *RunOptions) error {
 		glfw.WindowHint(glfw.Visible, glfw.True)
 	}
 
+	// Get our target monitor.
+	var monitor *Monitor
+
+	if monitorID := u.getInitWindowMonitor(); monitorID != glfw.DontCare {
+		monitor = monitors[monitorID]
+	} else {
+		m, err := initialMonitorByOS()
+		if err != nil {
+			return err
+		}
+		if m == nil {
+			m = glfw.GetPrimaryMonitor()
+		}
+
+		// GetPrimaryMonitor might return nil in theory (#1887).
+		if m == nil {
+			return errors.New("ui: no monitor was found at initialize")
+		}
+		for _, mon := range monitors {
+			if mon.m == m {
+				monitor = mon
+				break
+			}
+		}
+	}
+
+	u.setInitMonitor(monitor.m)
```

Comment:
Now `setInitMonitor` is called only here, we can extract the function's content here. Thoughts?

Reply:
Yes, that would work fine, providing the previous code didn't cause problems that I didn't find.

Reply:
NVM